lib/Models/Terria.ts

```diff
@@ -1206,34 +1232,32 @@ function generateInitializationUrl(
 }
 
 const loadInitSource = createTransformer(
-  (initSource: InitSource): Promise<JsonObject | undefined> => {
-    let promise: Promise<JsonValue | undefined>;
-
+  async (initSource: InitSource): Promise<JsonObject | undefined> => {
+    let jsonValue: JsonValue | undefined;
     if (isInitUrl(initSource)) {
-      promise = loadJson5(initSource.initUrl);
+      jsonValue = await loadJson5(initSource.initUrl);
     } else if (isInitOptions(initSource)) {
-      promise = initSource.options.reduce((previousOptionPromise, option) => {
-        return previousOptionPromise
-          .then(json => {
-            if (json === undefined) {
-              return loadInitSource(option);
-            }
-            return json;
-          })
-          .catch(_ => {
+      await initSource.options.reduce(async (previousOptionPromise, option) => {
+        try {
+          const json = await previousOptionPromise;
+          if (json === undefined) {
             return loadInitSource(option);
-          });
+          }
+          return json;
+        } catch (_) {
+          return loadInitSource(option);
+        }
       }, Promise.resolve<JsonObject | undefined>(undefined));
-    } else {
-      promise = Promise.resolve(initSource.data);
+    } else if (isInitData(initSource)) {
+      jsonValue = initSource.data;
+    } else if (isInitDataPromise(initSource)) {
+      jsonValue = (await initSource).data;
     }
 
-    return promise.then(jsonValue => {
-      if (isJsonObject(jsonValue)) {
-        return jsonValue;
-      }
-      return undefined;
-    });
+    if (jsonValue && isJsonObject(jsonValue)) {
+      return jsonValue;
+    }
+    return undefined;
```

Comment:
Asyncified + added support for `InitDataPromise` - which is just a `Promise` which resolves to `InitData`

Reply:
Still reviewing this